tests/abi/abi.c

```diff
@@ -0,0 +1,100 @@
+#include <stdint.h>
+#include <stdio.h>
+#include <stdlib.h>
+
+#include "for_each.h"
+#include "functions.h"
+
+#ifdef __x86_64__
+# include "platform_specific/x86_64.h"
+#elif __x86__
+# include "platform_specific/x86.h"
+#else
+ _Static_assert(0, "unsupported platform architecture");
+#endif
+
+/* LFSR generator */
+
+static uint64_t LFSR = 0xdc984ef2647af904ull;
+uint64_t getNextLFSR(void) {
+  uint32_t Iteration;
+  for (Iteration = 0; Iteration < 64; ++Iteration)
+    LFSR = (LFSR >> 1) | ((((LFSR >> 1) ^ (LFSR >> 2)) & 1) << 63);
+  return LFSR;
+}
+
+/* Helpers*/
+
+#define GET_1ST_HELPER(THIS_ONE, ...) THIS_ONE
+#define GET_1ST(THIS_ONE, ...) GET_1ST_HELPER(THIS_ONE, __VA_ARGS__)
+#define FIRST_REGISTER GET_1ST(REGISTER_LIST)
+
+#define STACK_VALUE_COUNT 32
+#define REGISTER_COUNT COUNT_ARGUMENTS(REGISTER_LIST)
+#define GENERATED_COUNT REGISTER_COUNT + STACK_VALUE_COUNT
+
+void dumpGenerated(REGISTER_C_TYPE *Generated) {
+  /* Dump registers*/
+  puts("Registers:");
+  {
+    uint32_t RegisterIndex = REGISTER_COUNT - 1;
+    #define DUMP_REGISTER(REGISTER) printf("  \"%s\": 0x%016lx\n", \
+                                           STRINGIFY(REGISTER), \
+                                           Generated[RegisterIndex--]);
```

Comment:
I'd use `do { } while (0)` syntax for multiline macros everywhere.

Reply:
[Done](https://github.com/revng/revng-qa/pull/4/commits/d4d3357bc16ea9b5813941ac8de36acd2afb65ed).